logfire-api/logfire_api/_internal/main.pyi

```diff
@@ -223,28 +223,10 @@ class Logfire:
             attributes: The arguments to include in the span and format the message template with.
                 Attributes starting with an underscore are not allowed.
         """
-    def instrument(self, msg_template: LiteralString | None = None, *, span_name: str | None = None, extract_args: bool | Iterable[str] = True, allow_generator: bool = False) -> Callable[[Callable[P, R]], Callable[P, R]]:
-        """Decorator for instrumenting a function as a span.
-
-        ```py
-        import logfire
-
-        logfire.configure()
-
-
-        @logfire.instrument('This is a span {a=}')
-        def my_function(a: int):
-            logfire.info('new log {a=}', a=a)
-        ```
-
-        Args:
-            msg_template: The template for the span message. If not provided, the module and function name will be used.
-            span_name: The span name. If not provided, the `msg_template` will be used.
-            extract_args: By default, all function call arguments are logged as span attributes.
-                Set to `False` to disable this, or pass an iterable of argument names to include.
-            allow_generator: Set to `True` to prevent a warning when instrumenting a generator function.
-                Read https://logfire.pydantic.dev/docs/guides/advanced/generators/#using-logfireinstrument first.
-        """
+    @overload
+    def instrument(self, msg_template: LiteralString | None = None, *, span_name: str | None = None, extract_args: bool | Iterable[str] = True, allow_generator: bool = False) -> Callable[[Callable[P, R]], Callable[P, R]]: ...
+    @overload
+    def instrument(self, func: Callable[P, R]) -> Callable[P, R]: ...
```

Comment:
@Kludex apparently stubgen doesn't work with docstrings and overloads?

Reply:
We need to add the flag `--include-docstrings`.

Reply:
Ah... Sorry. Got it.

Reply:
Sad.

Reply:
Maybe because of the wrong `func` name?

Reply:
What if the `func` becomes `msg_template`, does it work?

Reply:
drop that ruff rule